storage/api/http.go

```diff
@@ -40,44 +54,98 @@ func StoreFile(c echo.Context) error {
 
 	defer src.Close()
 
-	// Creat the path destination and mak sure we can copy the fie to the desired path.
-	filename := CreateUniqueFileName(file.Filename)
-	path := fmt.Sprintf("%s/%s", fileFolder, filename)
+	// Create the path destination and mak sure we can copy the fie to the desired path.
+	path, filename, err := storeFileToDisk(fileFolder, file.Filename, src)
 
-	dst, err := os.Create(path)
 	if err != nil {
 		return err
 	}
-	defer dst.Close()
 
-	// Copying the file.
-	if _, err = io.Copy(dst, src); err != nil {
-		return err
+	// Create the base response object in which we will append files.
+	response := FilesResponse{Files: []FileResponse{}}
+
+	if strings.HasSuffix(filename, ".zip") {
+		r, err := zip.OpenReader(path)
+
+		if err != nil {
+			return err
+		}
+
+		defer r.Close()
+
+		for _, f := range r.File {
+			uniqueFilenameFromZip, path := getUniqueNameAndPathFromFile(fileFolder, f.Name)
```

Comment:
Extract the logic to another file - `storageAPI` - as well as other helping function